File: 01/01.py
#!/usr/bin/env python3

import sys

with open(sys.argv[1]) as f:
    elves = f.read().strip().split('\n\n')
    calories = sorted(map(
        lambda elf: sum(map(
            lambda snack: int(snack),
            elf.split('\n')
        )),
        elves
    ))
    print(calories[-1])
    print(sum(calories[-3:]))

File: 01/01.rs
use std::env;
use std::fs::read_to_string;

fn read_lines(filename: &str) -> Vec<String> {
    read_to_string(filename).unwrap().lines().map(String::from).collect()
}

fn cals_for_elf(elf: &[String]) -> u32 {
    elf.iter().map(|s| s.parse::<u32>().unwrap()).sum::<u32>()
}

fn main() {
    let args: Vec<_> = env::args().collect();
    let mut calories: Vec<_> = read_lines(&args[1]).split(|l| l == "").map(cals_for_elf).collect();
    calories.sort();
    println!("{}", calories.last().unwrap());
    println!("{}", calories[(calories.len() - 3)..].iter().sum::<u32>())
}
